### PR TITLE
Manage CF and API custom domains by Serverless, separate dev and prod secrets

### PR DESCRIPTION
**Pull request by forabi**
(instead of having to manually configure these settings)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ script:
     -e TEST_DB_PORT=3306
     -e BRANCH=$TRAVIS_BRANCH
     -e COMMIT_ID=$TRAVIS_COMMIT
+    -e STAGE=production
```
Comment (wholesomedev) [inline .travis.yml]: STAGE could be misinterpreted as a "staging" environment (which is what just happened with me). A better name of this variable could be "ENVIRONMENT" or "ENV"

File: deploy.js [hunk 2]
```diff
@@ -51,7 +57,8 @@ async function main() {
   ];
   const deploymentCommands = [
     () => decryptSecrets(secrets, './secrets'),
-    'NODE_ENV=production yarn serverless deploy --stage production',
+    `yarn serverless create_domain --stage ${STAGE}`,
+    `NODE_ENV=production yarn serverless deploy --aws-s3-accelerate --stage ${STAGE}`,
```
Comment (wholesomedev) [inline deploy.js]: Oh Serverless actually calls it `stage`? Nevermind then I guess :)

File: serverless.yml [hunk 3]
```diff
@@ -63,3 +80,53 @@ package:
...
+              ViewerProtocolPolicy: 'redirect-to-https'
+              MinTTL: 0
+              AllowedMethods:
+                - GET
+                - HEAD
+                - OPTIONS
+                - POST
+                - PUT
+                - PATCH
+                - DELETE
```
Comment (wholesomedev) [inline serverless.yml]: I wonder if we can do `- *` here...

Comment (forabi) [reply serverless.yml]: Doesn't seem possible according to the docs https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-cachebehavior.html#cfn-cloudfront-distribution-cachebehavior-allowedmethods

**Pull request merged**